### PR TITLE
Develop pwa catchup

### PR DESCRIPTION
**Pull request by bhardwajaditya**
Upgrade to version 3.2.0

TODO: Modify [messageBoxAudioMessage.js](https://github.com/WideChat/Rocket.Chat/blob/develop/app/ui-message/client/messageBox/messageBoxAudioMessage.js) to  use latest changes.

### CODE REVIEW

**Comment by ear-dev** [comment]
@bhardwajaditya I'm just starting to look at this, and at first glance I see a merge error in the header bar, in the chatroom view on mobile.  

- The colors are black instead of white on the magnifying glass and the overflow menu.  
- The status indicator is doubled
- The back button is in its own row instead of inline with the other icons.

Also, do you know anything about that new, "You have joined a new direct message with... " banner at the end of the message list?  I'm not sure I understand if that is meant to go away or stay??

image

**Comment by bhardwajaditya** [comment]
> @bhardwajaditya I'm just starting to look at this, and at first glance I see a merge error in the header bar, in the chatroom view on mobile.
> 
> * The colors are black instead of white on the magnifying glass and the overflow menu.
> * The status indicator is doubled
> * The back button is in its own row instead of inline with the other icons.
> 
> Also, do you know anything about that new, "You have joined a new direct message with... " banner at the end of the message list? I'm not sure I understand if that is meant to go away or stay??
> 
> image

Ohh In that case @ear-dev  I need to perform a complete non regression test on this branch. Can you please check this test which is failing. It says some docker issue, don't know much about this one

**Comment by ear-dev** [comment]
@bhardwajaditya I don't think we have to worry about that docker issue.  I can look into it, but it's not related to the merge itself.  Are you able to fix the chatroom header?

**Comment by ear-dev** [comment]
@bhardwajaditya Looks like this issue is back in this PR: https://github.com/WideChat/Rocket.Chat/issues/198

Maybe when we find areas of code like this, that we have made Viasat specific, we should add a comment that might make it easier for you to catch in future catchup merges?  Would something like that help?

**Comment by ear-dev** [comment]
Can you also remove that little 'home' icon in the header please?  It seems like a new feature which we would only want to maintain in the web client, but not in the mobile view / PWA.

image

**Comment by ear-dev** [comment]
spacing in the header on the web client is off:

image

**Comment by bhardwajaditya** [comment]
> @bhardwajaditya Looks like this issue is back in this PR: #198
> 
> Maybe when we find areas of code like this, that we have made Viasat specific, we should add a comment that might make it easier for you to catch in future catchup merges? Would something like that help?

Yes will refactor this part to avoid future merge errors

**Comment by bhardwajaditya** [comment]
> spacing in the header on the web client is off:
> 
> image

@eric this looks fine in my browser. May be you can try clearing cache and then try this again.

**Comment by ear-dev** [comment]
@bhardwajaditya This is looking good.....  but I think I found another regression.  Did we lose this fix?  https://github.com/WideChat/Rocket.Chat/pull/226/files

Specifically it's working in public channels, but when in a DM it redirects to a new DM with the quoted user.  I specifically remember Shubham fixing this.  Seems like we need some method of flagging our code changes so that we don't just over write our fixes during these merges?

Do you mind taking a look please?  Thanks.

**Comment by ear-dev** [comment]
@bhardwajaditya We lost this one too I am afraid:  https://github.com/WideChat/Rocket.Chat/pull/227/files

Thanks.

File: app/ui/client/views/app/room.js [hunk 1]
```diff
@@ -883,7 +883,7 @@ Template.room.events({
 		const { msg } = messageArgs(this);
 		const link = msg.attachments ? msg.attachments[0].message_link : this.message_link;
 		const repliedMessageId = link.split('?msg=')[1];
-		const { pathname } = new URL(link);
+		const pathname = `/${ link.split('?msg=')[0].split('/').slice(-2)[0] }/${ link.split('?msg=')[0].split('/').slice(-1)[0] }`;
```
Comment (ear-dev) [inline app/ui/client/views/app/room.js]: @bhardwajaditya This is not actually the fix that was merged into our develop_pwa branch...  please see:  https://github.com/WideChat/Rocket.Chat/pull/226/files

**Comment by ear-dev** [comment]
@bhardwajaditya One more thing please......  can you remove the text from the status indicator in the chat room header please?  We currently don't use it, and I think it looks cleaner.   Thanks.

image

File: app/ui/client/views/app/room.js (hunk 1 above)
Comment (bhardwajaditya) [reply app/ui/client/views/app/room.js]: Ohh I am sorry. I mixed up the commit.

**Comment by ear-dev** [comment]
@bhardwajaditya This looks good now.  We may find other issues when we start using it, but I think I can merge this now.  Thanks!

**Pull request merged**